decidim-proposals/app/commands/decidim/proposals/vote_proposal.rb

```diff
@@ -26,7 +26,7 @@ def call
         return broadcast(:invalid) unless vote.valid?
 
         ActiveRecord::Base.transaction do
-          proposal.with_lock do
+          @proposal.with_lock do
```

Comment:
Why are we locking the proposal if we are inserting the votes inside `decidim_proposals_proposal_votes`? is it to protect the `proposal_votes_count` counter cache? what can allow a user to vote many times?

Reply:
Ok, we've been commenting it with @MarcReniu and we understand that the `select for update` is to protect for the case that two users vote at the very same moment and the vote limit is exceeded.